### PR TITLE
Fix s3 script

### PR DESCRIPTION
**Pull request by ILpinto**
Fix input method delete_s3_velero_bucket.py

### CODE REVIEW

**Comment by omrirh** [comment]
!lgtm

**Comment by omrirh** [comment]
needs rebase; except that lgtm

File: aws/delete_s3_velero_bucket.py [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+        if verify_cluster_matches_velero_infrastructure_name(
+            cluster_name=cluster,
+            bucket_name=bucket["Name"],
+        ):
+            delete_all_objects_from_s3_folder(
+                bucket_name=bucket["Name"],
+            )
+
+            delete_bucket(bucket_name=bucket["Name"])
+            # A cluster is mapped to only one bucket, so don't check any remaining buckets
```
Comment (myakove) [inline aws/delete_s3_velero_bucket.py:67]: in this case don't you want to check that `len(velero_bucket_list) == 1` ?

File: aws/delete_s3_velero_bucket.py [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+    parser = argparse.ArgumentParser(description="Delete an s3 velero cluster bucket")
+    parser.add_argument(
+        "-c",
+        "--cluster_name",
+        help=""" Either set the S3_CLUSTER environment variable or
+                    Pass the cluster value at the command line when the script is invoked with the -c|--cluster
+                """,
+    )
+    cluster_name = parser.parse_args().cluster_name
+    if len(found_env) < len(REQUIRE_ENV):
```
Comment (myakove) [inline aws/delete_s3_velero_bucket.py:220]: Please check this before setting anything else (move it to after line 210)

**Review by myakove — CHANGES_REQUESTED** [review]

File: aws/delete_s3_velero_bucket.py (hunk 2 above)
Comment (ILpinto) [reply aws/delete_s3_velero_bucket.py:220]: Fixed

File: aws/delete_s3_velero_bucket.py (hunk 1 above)
Comment (ILpinto) [reply aws/delete_s3_velero_bucket.py:67]: it done in: line 52
velero_bucket_list = get_velero_buckets()
    if not velero_bucket_list:
        LOGGER.info("No velero buckets found")
        return

File: aws/README.md [hunk 3]
```diff
@@ -1 +1,4 @@
-#AWS tools
+# AWS tools
+
+## Delete S3 velero bucket
+This script deletes a bucket associated with a cluster after the cluster is uninstalled.
```
Comment (myakove) [inline aws/README.md:4]: This script will delete the bucket regardless the cluster state (installed/uninstall) right? 
so `after the cluster is uninstalled.` is not right.

File: aws/delete_s3_velero_bucket.py [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+
+    velero_bucket_list = get_velero_buckets()
+    if not velero_bucket_list:
+        LOGGER.info("No velero buckets found")
+        return
+
+    for bucket in velero_bucket_list:
+        if verify_cluster_matches_velero_infrastructure_name(
+            cluster_name=cluster,
+            bucket_name=bucket["Name"],
```
Comment (myakove) [inline aws/delete_s3_velero_bucket.py:60]: save `bucket["Name"]` and reuse it

File: aws/delete_s3_velero_bucket.py [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+
+
+def delete_bucket(bucket_name: str) -> None:
+    """
+    Delete velero bucket
+
+    Args:
+        bucket_name: the bucket name
+    """
+    LOGGER.info(f"Delete cluster bucket {bucket_name}")
```
Comment (myakove) [inline aws/delete_s3_velero_bucket.py:80]: remove the word `cluster` from the logger, this delete bucket

File: aws/delete_s3_velero_bucket.py [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+    response = S3_CLIENT.delete_bucket(Bucket=bucket_name)
+
+    response_http_status_code = response["ResponseMetadata"]["HTTPStatusCode"]
+    if response_http_status_code == HTTPStatus.NO_CONTENT:
+        LOGGER.info(f"Cluster bucket '{bucket_name}' deleted successfully")
+        return
+
+    elif response_http_status_code == HTTPStatus.NOT_FOUND:
+        LOGGER.info(f"Bucket '{bucket_name}' not found\n")
```
Comment (myakove) [inline aws/delete_s3_velero_bucket.py:89]: why the `\n` is needed in the logger?

File: aws/delete_s3_velero_bucket.py [hunk 7]
```diff
@@ -0,0 +1,234 @@
...
+
+def get_velero_infrastructure_name(bucket_name: str) -> Union[str, None]:
+    """
+    Get the velero bucket infrastructure name
+
+    Args:
+        bucket_name: The name of the cluster bucket to delete
+
+    Returns:
+        velero infrastructure name or None
```
Comment (myakove) [inline aws/delete_s3_velero_bucket.py:124]: `str or None: velero infrastructure name if found else None`

File: aws/delete_s3_velero_bucket.py [hunk 8]
```diff
@@ -0,0 +1,234 @@
...
+) -> bool:
+    """
+    Get the velero bucket infrastructure name and compare it with the cluster
+
+    Args:
+        cluster_name: The cluster name
+        bucket_name: The bucket name
+
+    Returns:
+        True if the cluster matches a velero infrastructure name, False otherwise
```
Comment (myakove) [inline aws/delete_s3_velero_bucket.py:147]: `Returns:` convention is 
type: description

Please fix.

File: aws/delete_s3_velero_bucket.py [hunk 9]
```diff
@@ -0,0 +1,234 @@
...
+    for env_var in REQUIRE_ENV:
+        if os.getenv(env_var):
+            found_env.append(env_var)
+    if len(found_env) < len(REQUIRE_ENV):
+        LOGGER.error(
+            f"Missing environment variable found: {found_env}, require:{REQUIRE_ENV}"
+        )
+        sys.exit(1)
+
+    parser = argparse.ArgumentParser(description="Delete an s3 velero cluster bucket")
```
Comment (myakove) [inline aws/delete_s3_velero_bucket.py:217]: I think you can set args as required and set default to them
then you do not need to check the REQUIRE_ENV, set arg for each as required and have default for each

also if `--cluster_name` is also required please set it as required

**Review by myakove — CHANGES_REQUESTED** [review]

File: aws/delete_s3_velero_bucket.py (hunk 9 above)
Comment (ILpinto) [reply aws/delete_s3_velero_bucket.py]: using click and all args are required, AWS parameters are set as env variable and set as default.

File: aws/delete_s3_velero_bucket.py (hunk 1 above)
Comment (myakove) [reply aws/delete_s3_velero_bucket.py:65]: I mean, you exit here inside the loop after the first match, but what if you have more than one bucket in the list?

File: requirements.txt [hunk 10]
```diff
@@ -0,0 +1,2 @@
+boto3
+click~=8.1.3
```
Comment (myakove) [inline requirements.txt:2]: Any reason to pin click to version?

File: aws/delete_s3_velero_bucket.py (hunk 1 above)
Comment (ILpinto) [reply aws/delete_s3_velero_bucket.py:65]: Velero bucket are created as backup for the cluster meta date, we will have one bucket for cluster. We loop to file the bucket by cluster name, clean the objects and delete it.

**Comment by omrirh** [comment]
!lgtm

**Review by myakove — APPROVED** [review]

**Pull request merged**